### PR TITLE
Nintendo Switch support

### PR DESCRIPTION
**Pull request by fgsfdsfgs**
Adds Nintendo Switch support to both waf and cmake scripts. Complementary to FWGS/xash3d-fwgs#1238.
Building the libraries for `nswitch` requires mostly same dependencies as building the engine for it.

**Building with waf:**
```
./waf configure -T release --nswitch
./waf build
```

**Building with CMake:**
```
mkdir build && cd build
aarch64-none-elf-cmake -G"Unix Makefiles" -DCMAKE_PROJECT_HLSDK-PORTABLE_INCLUDE="$DEVKITPRO/portlibs/switch/share/SolderShim.cmake" ..
make -j
```

### CODE REVIEW

File: cmake/LibraryNaming.cmake [hunk 1]
```diff
@@ -76,7 +79,7 @@ elseif(XASH_X86)
 		set(BUILDARCH "i386")
 	endif()
 elseif(XASH_ARM AND XASH_64BIT)
-	set(BUILDARCH "arm64")
+	set(BUILDARCH "aarch64")
```
[the remark this answers is not in the document]
Comment (a1batross) [reply cmake/LibraryNaming.cmake:82]: No, it wasn't a typo!

Please, revert these. It's an engine bug, we shouldn't enforce ever change naming without a reason.

**Review by a1batross — CHANGES_REQUESTED** [review]

File: cmake/LibraryNaming.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (fgsfdsfgs) [reply cmake/LibraryNaming.cmake]: Reverted in b01237f.

**Review by a1batross — APPROVED** [review]

**Comment by a1batross** [comment]
сс @nekonomicon can you take a look and merge if it's ok?

**Comment by nekonomicon** [comment]
@fgsfdsfgs can you add build instructions to readme yet?

**Comment by a1batross** [comment]
@nekonomicon don't you find README a bit cluttered for that? Better to decouple all the instructions into Documentation/ folder. Or have a self-documenting build tool, like Waf. :)

**Comment by fgsfdsfgs** [comment]
> @fgsfdsfgs can you add build instructions to readme yet?

Done.

**Comment by nekonomicon** [comment]
LGTM

**Pull request merged**